applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -164,11 +158,80 @@ void TwoFluidNavierStokes<TElementData>::CalculateLocalSystem(
                     ComputeGaussPointEnrichmentContributions(data, Vtot, Htot, Kee_tot, rhs_ee_tot);
                 }
 
+                Matrix int_shape_function, int_shape_function_enr_neg, int_shape_function_enr_pos;
+                GeometryType::ShapeFunctionsGradientsType int_shape_derivatives;
+                Vector int_gauss_pts_weights;
+                std::vector< array_1d<double,3> > int_normals_neg;
+
+                if (rCurrentProcessInfo[SURFACE_TENSION] || rCurrentProcessInfo[MOMENTUM_CORRECTION]){
+                    ComputeSplitInterface(
+                        data,
+                        int_shape_function,
+                        int_shape_function_enr_pos,
+                        int_shape_function_enr_neg,
+                        int_shape_derivatives,
+                        int_gauss_pts_weights,
+                        int_normals_neg,
+                        p_modified_sh_func);
+                }
+
+                if (rCurrentProcessInfo[MOMENTUM_CORRECTION]){
+                    MatrixType lhs_acc_correction = ZeroMatrix(NumNodes*NumNodes,NumNodes*NumNodes);
+
+                    double positive_density;
+                    double negative_density;
+
+                    GeometryType::Pointer p_geom = this->pGetGeometry();
```

Comment:
```suggestion
                    const auto& r_geom = this->GetGeometry();
```
As far as I see below you don't need a pointer. I'd use a reference to avoid the derreference operator each time you access a node.

Reply:
Done.